WooCommerce/Classes/ViewRelated/Products/Variations/Bulk Update/BulkUpdateViewModel.swift

```diff
@@ -59,26 +76,99 @@ final class BulkUpdateViewModel {
 
                 if let error = error {
                     self.syncState = .error
 
                     DDLogError("⛔️ Error synchronizing product variations: \(error)")
                 } else {
-                    self.syncState = .synced
+                    self.configureResultsControllerAndFetchData { [weak self] error in
+                        if error != nil {
+                            self?.syncState = .error
+                        } else {
+                            self?.syncState = .synced
+                        }
+                    }
                 }
             }
 
         storesManager.dispatch(action)
     }
+
+    /// Provides the view model with all user facing data of the option for updating the regular price
+    ///
+    func viewModelForDisplayingRegularPrice() -> ValueOneTableViewCell.ViewModel {
+        return ValueOneTableViewCell.ViewModel(text: Localization.updateRegularPriceTitle,
+                                               detailText: detailTextforPrice(\ProductVariation.regularPrice))
+    }
+
+    /// Returns the detail text of the option for updating a price
+    ///
+    private func detailTextforPrice(_ priceKeyPath: KeyPath<ProductVariation, String?>) -> String {
+        guard let bulkUpdateFormModel = bulkUpdateFormModel else {
+            return ""
+        }
+        switch bulkUpdateFormModel.bulkValueOf(priceKeyPath) {
+        case .none:
+            return Localization.noneValue
+        case .mixed:
+            return Localization.mixedValue
+        case let .value(price):
+            return formatPriceString(price)
+        }
+    }
+
+    /// It formats a price `String` according to the current price settings.
+    ///
+    private func formatPriceString(_ price: String) -> String {
+        let currencyCode = currencySettings.currencyCode
+        let currency = currencySettings.symbol(from: currencyCode)
+
+        return currencyFormatter.formatAmount(price, with: currency) ?? ""
+    }
+
+    /// Setup the results controller
+    ///
+    private func configureResultsControllerAndFetchData(onCompletion: @escaping (Error?) -> Void) {
+        resultsController.onDidChangeContent = { [weak self] in
+            guard let self = self else { return }
+            self.updateDataModel()
+            onCompletion(nil)
+        }
+
+        do {
+            try resultsController.performFetch()
+            updateDataModel()
+            onCompletion(nil)
+        } catch {
+            DDLogError("⛔️ Error fetching product variations: \(error)")
+            onCompletion(error)
+        }
+    }
+
+    /// When new data are available we should update them and then update the syncSate
+    ///
+    private func updateDataModel() {
+        // Limit the variations to `Constants.numberOfObjects`, since it is the limit of th bulk update API
+        let candidatesForUpdate = Array(resultsController.fetchedObjects.prefix(Constants.numberOfObjects))
```

Comment:
Are you doing this just to be on the safe side? Because if we are only fetching 100 variations, would this ever have any effect?

Reply:
I am adding this as an extra precaution in the case the previous product variations screen syncs the next page after the bulk update finishes the sync, not likely but with the right networking conditions and server load it might happen. I didn't want to rely on network calls timing for the expected behaviour.